src/constraints.jl

```diff
@@ -53,7 +55,11 @@ it supports `F`-in-`S` constraints but it cannot add the constraint(s) in its
 current state.
 """
 function addconstraint!(model::ModelLike, func::AbstractFunction, set::AbstractSet)
-    throw(UnsupportedConstraint{typeof(func), typeof(set)}())
+    if supports(model, typeof(func), typeof(set))
+        throw(UnsupportedConstraint{typeof(func), typeof(set)}())
+    else
+        throw(CannotAddConstraint{typeof(func), typeof(set)}())
```

Comment:
This would read better IMO as `CannotAddConstraintTryReset`.

Reply:
The try reset can be in the doc or error message but I don't see why it would help to add it in the name. It is for every cannoterror, not just cannotaddconstraint

Reply:
I would argue that `TryReset` should be in the name of all the "cannot" types.

Reply:
And the super type is CannotErrorTryReset ?

Reply:
That's one option.